ARCHITECTURE.md

```diff
@@ -0,0 +1,52 @@
+# NUMA Resources Operator architecture
+
+## concepts
+
+The NUMA Resources Operator is a pilot operator to deploy all the components needed for the out-of-tree NUMA aware scheduler:
+1. NodeResourceTopology API (out-of-tree CRD-based API)
+2. ResourceTopologyExporter (RTE) as topology info provider
+3. The out-of-tree secondary scheduler with the NodeResourceTopologyMatch plugins enabled
+
+In addition, the operator repo includes a extensive e2e test suite (so-called "serial suite")
+
+## building blocks
+
+The operator is built on top of two main community-based projects: [the deployer toolkit](https://github.com/k8stopologyawareschedwg/deployer)
+and [the resource topology exporter](https://github.com/k8stopologyawareschedwg/resource-topology-exporter).
+
+The deployer toolkit provides functionalities to process the YAML manifests (or their golang object representation) to install all the components
+in the NUMA-aware stack. It's the core on which the operator builds on, adding operator-specific customizations.
+
+The resource-topology-exporter is vendored to enable the "bundled operand" pattern (see below). The operand reuses most of the RTE code, just
+replacing the main entry point.
+
+Logically, the operator manages three operands (API, RTE, scheduler) but it depends on just a single artifact for the operand (the scheduler),
+while the other two operands (API, RTE) are bundled in the same operator image.
+
+## bundled operand
+
+Instead of consuming a separate image like we do for the secondary scheduler, we bundle the RTE operand adding the binary in the operator image,
+which has multiple entry points.
+
+This decision was taken in order to minimize the number of artifacts and narrow down as much as possible the dependency chain.
+
+We foresee RTE as the first operand to be replaced, and in general is the easiest to replace. `nfd-topology-updater` is already part of NFD
+and a supported component. On the other hand, the secondary scheduler is likely to stick around for a while more - and that's also easier to
+support because secondary schedulers are not uncommon in the kubernetes ecosystem.
```

Comment:
just to confirm, is it intended to be "not uncommon"?

Reply:
yes. "not uncommon" in the sense that sometimes you see secondary schedulers, but not often.